WordPress/src/main/java/org/wordpress/android/ui/sitecreation/previews/SiteCreationPreviewFragment.kt

```diff
@@ -214,10 +223,44 @@ class SiteCreationPreviewFragment : SiteCreationBaseFormFragment(),
 
     private fun updateLoadingLayout(progressUiState: SitePreviewFullscreenProgressUiState) {
         progressUiState.apply {
-            uiHelpers.setTextOrHide(fullscreenProgressLayout.findViewById(R.id.progress_text), loadingTextResId)
+            val newText = uiHelpers.getTextOfUiString(loadingTextView.context, loadingTextResId)
+            AppLog.d(AppLog.T.MAIN, "Changing text - animation: $animate")
+            if (animate) {
+                updateLoadingTextWithFadeAnimation(newText)
+            } else {
+                loadingTextView.text = newText
+            }
         }
     }
 
+    private fun updateLoadingTextWithFadeAnimation(newText: String) {
+        val animationDuration = AniUtils.Duration.SHORT
+        val animations = ArrayList<Animator>()
+        val fadeOut = AniUtils.getFadeOutAnim(loadingTextLayout, animationDuration, View.VISIBLE)
+        val fadeIn = AniUtils.getFadeInAnim(loadingTextLayout, animationDuration)
+
+        // update the text when the view isn't visible
+        fadeIn.addListener(object : AnimatorListener {
+            override fun onAnimationRepeat(animation: Animator?) {}
+
+            override fun onAnimationEnd(animation: Animator?) {}
+
+            override fun onAnimationCancel(animation: Animator?) {}
+
+            override fun onAnimationStart(animation: Animator) {
+                loadingTextView.text = newText
+            }
+        })
```

Comment:
Hi @malinajirka can this implementation below work since we don't need the other methods of the listener? Let me know 😄 
```suggestion
             fadeIn.addListener(object : AnimatorListenerAdapter() {
            override fun onAnimationStart(animation: Animator?) {
                loadingTextView.text = newText
            }
        })
```

Reply:
Hi Joel ;), the methods don't have a default implementation so I believe we need to provide some implementation. Even when the implementation is just an empty method.  AS is complaining when we remove them. Wdyt?

Reply:
Okay understood. Correct me if I am wrong. But doesn't the `AnimatorListenerAdapter` have the default implementation and we would just override the method we want to utilize?

Also for AS, what kind of warning are you seeing? This how mine looks. 

<img width="831" alt="Screen Shot 2020-04-07 at 9 18 09 AM" src="https://user-images.githubusercontent.com/1509205/78685509-737ad700-78b7-11ea-9b3a-4a0aba0acf66.png">

Reply:
🤦‍♂ I've totally missed the "AnimatorListenerAdapter" - TIL. You are absolutely right and this is a nice clean up ;)! Fixed in 02d9838962c8611bda746acafb99eba97b05a6d9